extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/RuntimeBeanRepository.java

```diff
@@ -163,8 +165,13 @@ public <T> T findSingleByType(Class<T> type) {
             } else {
                 handles = container.listAll(type);
             }
+
             if (handles.size() > 0) {
-                return handles.get(0).get();
+                if (handles.stream().filter(h -> !h.getBean().getKind().equals(InjectableBean.Kind.SYNTHETIC)).count() > 0) {
```

Comment:
What's the significance of the `SYNTHETIC` beans kind where we can just take the first one? Is that safe if multiple results are returned?

Reply:
Synthetic beans are usually provided by extensions. I'm thinking another way with resorting the handels?

The orignal `handles` is sorted by `priority` and when `priority` is equal, the position is random. How about to introduce a new `Comparator` if `priority` is the same, the default bean is win?

Reply:
> introduce a new Comparator if priority is the same, the default bean is win?

Yeah, sounds ok.

Probably faster than the alternative, which would be to collect all priorities in a set and then check the size.

Reply:
OK, I updated with the new `Comparator`.